pages/data-providers/data-provider.jsx

```diff
@@ -1,7 +1,125 @@
-import React from 'react'
+import React, { useEffect } from 'react'
+import { useRouter } from 'next/router'
+import { useLocalStore, useObserver } from 'mobx-react-lite'
 
 import DataProviderPageTemplate from 'templates/data-provider'
+import useDebouncedEffect from 'hooks/use-debounced-effect'
 
-const DataProviderPage = () => <DataProviderPageTemplate />
+// TODO: Pages should be just a thin layer of NextJS.
+//       Find a better place for it.
+const useDiscoveryStore = (defaultDataProviderUrl) => {
+  const store = useLocalStore(() => ({
+    query: defaultDataProviderUrl,
+    request: null,
+    loading: false,
```

Comment:
Maybe `@computed get loading() { return this.request == null }`?

Reply:
I will create is a class and will add it.